docs/reference/QueryClient.md

```diff
@@ -178,6 +179,22 @@ const data = queryClient.getQueryData({ queryKey })
 - `data: TQueryFnData | undefined`
   - The data for the cached query, or `undefined` if the query does not exist.
 
+## `queryClient.ensureQueryData`
+
+`ensureQueryData` is an asynchronous function that can be used to get an existing query's cached data. If the query does not exist, `queryClient.fetchQuery` will be called and its results returned.
+
+```tsx
+const data = queryClient.ensureQueryData({ queryKey, queryFn })
+```
+
+**Options**
+
+- The options for `ensureQueryData` are exactly the same as those of [`fetchQuery`](#queryclientfetchquery).
```

Comment:
except that `queryKey` is required maybe?

Reply:
[b0537ed](https://github.com/TanStack/query/pull/4544/commits/b0537ed35af9dcdbb08f0765cb3bbc67499048f4)